components/core/tests/test-ParserWithUserSchema.cpp

```diff
@@ -79,56 +93,55 @@ void decompress(std::string archive_dir, std::string output_dir) {
 }
 
 TEST_CASE("Test error for missing schema file", "[LALR1Parser][SchemaParser]") {
-    std::string file_path = "../tests/test_schema_files/missing_schema.txt";
-    std::string file_name = boost::filesystem::weakly_canonical(file_path).string();
+    auto const file_path = test_schema_dir / "missing_schema.txt";
     REQUIRE_THROWS_WITH(
-            generate_schema_ast(file_path),
-            "Failed to read '" + file_path + "', error_code="
+            generate_schema_ast(file_path.string()),
+            "Failed to read '" + file_path.string() + "', error_code="
                     + std::to_string(static_cast<int>(log_surgeon::ErrorCode::FileNotFound))
     );
 }
 
 TEST_CASE("Test error for empty schema file", "[LALR1Parser][SchemaParser]") {
-    std::string file_path = "../tests/test_schema_files/empty_schema.txt";
+    auto const file_path = test_schema_dir / "empty_schema.txt";
     REQUIRE_THROWS_WITH(
-            generate_schema_ast(file_path),
+            generate_schema_ast(file_path.string()),
             "Schema:1:1: error: empty file\n"
             "          \n"
             "^\n"
     );
 }
 
 TEST_CASE("Test error for colon missing schema file", "[LALR1Parser][SchemaParser]") {
-    std::string file_path = "../tests/test_schema_files/colon_missing_schema.txt";
+    auto const file_path = test_schema_dir / "colon_missing_schema.txt";
     REQUIRE_THROWS_WITH(
-            generate_schema_ast(file_path),
+            generate_schema_ast(file_path.string()),
             "Schema:3:4: error: expected '>',':','AlphaNumeric' before ' ' token\n"
             "          int [0-9]+\n"
             "             ^\n"
     );
 }
 
 TEST_CASE("Test error for multi-character tokens in schema file", "[LALR1Parser][SchemaParser]") {
-    std::string file_path = "../tests/test_schema_files/schema_with_multicharacter_token_error.txt";
+    auto const file_path = test_schema_dir / "schema_with_multicharacter_token_error.txt";
     REQUIRE_THROWS_WITH(
-            generate_schema_ast(file_path),
+            generate_schema_ast(file_path.string()),
             "Schema:2:11: error: expected ':' before ' ' token\n"
             "          delimiters : \\r\\n\n"
             "                    ^\n"
     );
 }
 
 TEST_CASE("Test creating schema parser", "[LALR1Parser][SchemaParser]") {
-    generate_schema_ast("../tests/test_schema_files/easy_schema.txt");
+    generate_schema_ast((test_schema_dir / "easy_schema.txt").string());
```

Comment:
Let's break the line into two lines for better readability. Ditto for other lines below
```suggestion
    auto const schema_file_path{test_schema_dir / "easy_schema.txt"};
    generate_schema_ast(schema_file_path.string());
```